builder/qemu/step_wait_guest_address.go

```diff
@@ -18,12 +18,16 @@ import (
 
 // This step waits for the guest address to become available in the network
 // bridge, then it sets the guestAddress state property.
-type stepWaitGuestAddress struct{}
+type stepWaitGuestAddress struct {
+	timeout time.Duration
+}
 
 func (s *stepWaitGuestAddress) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
 	qmpMonitor := state.Get("qmp_monitor").(*qmp.SocketMonitor)
+	ctx, cancel := context.WithTimeout(ctx, s.timeout)
```

Comment:
Line 41, there is a `case <-time.After(10 * time.Second):` which will cap this timeout to max 10s, do we want to keep this ?

Reply:
There are two things happening:

1. Every 10s we check if the guest IP address has appeared in the bridge. When it does, we return successfully.
2. When the timeout expires we give up and return an error.

Reply:
Actually this can be made a tiny bit more readable by replacing the `break` statement with the `return` statement. I will re-submit a patch in a bit.

Reply:
I've just force-pushed this change.

Reply:
Oh I understand it now; thanks for explaining/fixing ! :)